whylog/teacher/__init__.py

```diff
@@ -69,6 +98,20 @@ def _verify(self):
         """
         pass
 
+    def _prepare_rule(self):
+        """
+        :return UserRuleIntent
+        """
+
+    def get_rule(self):
+        """
+        Creates rule for Front, that will be shown to user
+        """
+        pass
+
+    def test_rule(self):
```

Comment:
Please don't add methods which do nothing, unless you need to define an interface, but in such case the method should be clearly documented.